hcl2template/types.packer_config.go

```diff
@@ -239,11 +218,37 @@ func (cfg *PackerConfig) getCoreBuildProvisioners(source SourceBlock, blocks []*
 	return res, diags
 }
 
+func (cfg *PackerConfig) getProvisioner(source SourceBlock, pb *ProvisionerBlock, ectx *hcl.EvalContext) (packer.Provisioner, hcl.Diagnostics) {
+	provisioner, moreDiags := cfg.startProvisioner(source, pb, ectx)
+	if moreDiags.HasErrors() {
+		return nil, moreDiags
+	}
+	// If we're pausing, we wrap the provisioner in a special pauser.
+	if pb.PauseBefore != 0 {
+		provisioner = &packer.PausedProvisioner{
+			PauseBefore: pb.PauseBefore,
+			Provisioner: provisioner,
+		}
+	} else if pb.Timeout != 0 {
+		provisioner = &packer.TimeoutProvisioner{
+			Timeout:     pb.Timeout,
+			Provisioner: provisioner,
+		}
+	}
+	if pb.MaxRetries != 0 {
+		provisioner = &packer.RetriedProvisioner{
+			MaxRetries:  pb.MaxRetries,
+			Provisioner: provisioner,
+		}
+	}
+	return provisioner, nil
+}
+
 // getCoreBuildProvisioners takes a list of post processor block, starts
 // according provisioners and sends parsed HCL2 over to it.
-func (cfg *PackerConfig) getCoreBuildPostProcessors(source SourceBlock, blocks []*PostProcessorBlock, ectx *hcl.EvalContext, generatedVars map[string]string) ([]packer.CoreBuildPostProcessor, hcl.Diagnostics) {
```

Comment:
I think we still need the `generatedVars` but may be as a `[]string` just to set them as cty unknown for now.

Reply:
This is only for JSON interpolation of build variables, we don't need them here anymore because we are not using that to interpolate such variables.

Reply:
ah yes you are right !